staging/src/k8s.io/client-go/util/certificate/certificate_manager.go

```diff
@@ -421,6 +446,10 @@ func (m *manager) rotateCerts() (bool, error) {
 		return false, nil
 	}
 
+	if m.certificateRotation != nil && m.cert != nil {
+		m.certificateRotation.Observe(m.clock.Now().Sub(m.cert.Leaf.NotBefore).Seconds())
```

Comment:
`m.cert` has to be accessed with a mutex (see `m.updateCached`)
maybe `updateCached` can just return the old cert?

Reply:
Good idea, done.